<launch>

  <!-- Default values -->
  <arg name="start_rviz" default="true"/>
  <arg name="map_config" default="$(find stargazer_ros_tool)/param/map.yaml"/>
  <arg name="landmark_topic" default="/landmarks"/>
  <arg name="map_frame_id" default="stargazer" />

  <node pkg="stargazer_ros_tool" type="landmark_visualizer" name="landmark_visualizer" output="screen" required="true">
    <param name="map_config" value="$(arg map_config)"/>
    <param name="landmark_topic" value="$(arg landmark_topic)"/>
    <param name="rate" value="0.2"/>
    <param name="map_frame_id" value="$(arg map_frame_id)" />
  </node>

  <!-- Start an rviz node with a custom configuration for the viewpoint, map_server, trajectory, laser scans, etc -->
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find stargazer_ros_tool)/res/stargazer.rviz" if="$(arg start_rviz)"/>

</launch>
